go-controller/hybrid-overlay/pkg/controller/master.go

```diff
@@ -381,10 +382,12 @@ func (m *MasterController) setupHybridLRPolicySharedGw(nodeSubnets []*net.IPNet,
 			Match:    grMatchStr,
 		}
 
-		if err := libovsdbops.CreateOrUpdateLogicalRouterPolicyWithPredicate(m.nbClient, ovntypes.OVNClusterRouter, &grLogicalRouterPolicy, func(item *nbdb.LogicalRouterPolicy) bool {
-			return item.Priority == grLogicalRouterPolicy.Priority &&
-				item.ExternalIDs["name"] == grLogicalRouterPolicy.ExternalIDs["name"]
-		}); err != nil {
+		// TODO(trozet): evaluate if safe to update all fields
+		if err := libovsdbops.CreateOrUpdateLogicalRouterPolicyWithPredicate(m.nbClient, ovntypes.OVNClusterRouter,
+			&grLogicalRouterPolicy, func(item *nbdb.LogicalRouterPolicy) bool {
+				return item.Priority == grLogicalRouterPolicy.Priority &&
+					item.ExternalIDs["name"] == grLogicalRouterPolicy.ExternalIDs["name"]
+			}, &logicalRouterPolicy.Nexthops, &logicalRouterPolicy.Match); err != nil {
```

Comment:
This is equivalent, as we were updating constant fields and neither NextHops or Match could be empty.